ql/src/java/org/apache/hadoop/hive/ql/stats/StatsUtils.java

```diff
@@ -454,6 +468,27 @@ private static Statistics collectStatistics(HiveConf conf, PrunedPartitionList p
     return stats;
   }
 
+  private static Map<String, Long> getValidBasicStatsFromStorageHandler(Table table) {
+
+    Map<String, String> storageHandlerBasicStatsMap =
+        (table.isNonNative() && table.getStorageHandler().canProvideBasicStatistics()) ? table.getStorageHandler()
+            .getBasicStatistics(Partish.buildFor(table)) : null;
+
+    try {
+      if (storageHandlerBasicStatsMap != null &&
+          Long.parseLong(storageHandlerBasicStatsMap.get(StatsSetupConst.ROW_COUNT)) > 0) {
```

Comment:
Does it mean that an empty table has no stats?

Reply:
Yes, snapshots are not created for a new table until data is inserted. As a result, stats are not available.

The following line turns out to be null for the new table.
` if (table.currentSnapshot() != null) `

https://github.com/apache/hive/blob/master/iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java#L316